packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/ExtractIconFromDataTests.m

```diff
@@ -0,0 +1,175 @@
+  XCTAssertNotNil(resultImage);
+  [mockImageClass stopMocking];
+}
+
+- (void)testExtractIconFromDataAssetNoScaling {
+  FLTGoogleMapMarkerController *instance = [[FLTGoogleMapMarkerController alloc] init];
+  NSObject<FlutterPluginRegistrar> *mockRegistrar =
+      OCMStrictProtocolMock(@protocol(FlutterPluginRegistrar));
+  id mockImageClass = OCMClassMock([UIImage class]);
+  UIImage *testImage = [self createOnePixelImage];
+
+  OCMStub([mockRegistrar lookupKeyForAsset:@"fakeImageNameKey"]).andReturn(@"fakeAssetKey");
+  OCMStub(ClassMethod([mockImageClass imageNamed:@"fakeAssetKey"])).andReturn(testImage);
+
+  NSArray *iconData = @[ @"asset", @"fakeImageNameKey", @"noScaling", @2 ];
+
+  UIImage *resultImage = [instance extractIconFromData:iconData registrar:mockRegistrar];
+  XCTAssertNotNil(resultImage);
+  [mockImageClass stopMocking];
+}
+
+- (void)testExtractIconFromDataBytesAuto {
+  FLTGoogleMapMarkerController *instance = [[FLTGoogleMapMarkerController alloc] init];
+  NSObject<FlutterPluginRegistrar> *mockRegistrar =
+      OCMStrictProtocolMock(@protocol(FlutterPluginRegistrar));
+  UIImage *testImage = [self createOnePixelImage];
+  NSData *pngData = [self convertImageToPNGData:testImage];
+  XCTAssertNotNil(pngData);
+
+  FlutterStandardTypedData *typedData = [FlutterStandardTypedData typedDataWithBytes:pngData];
+
+  NSArray *iconData = @[ @"bytes", typedData, @"auto", @2 ];
+
+  UIImage *resultImage = [instance extractIconFromData:iconData registrar:mockRegistrar];
+  XCTAssertNotNil(resultImage);
+}
+
+- (void)testExtractIconFromDataBytesAutoAndSize {
+  FLTGoogleMapMarkerController *instance = [[FLTGoogleMapMarkerController alloc] init];
+  NSObject<FlutterPluginRegistrar> *mockRegistrar =
+      OCMStrictProtocolMock(@protocol(FlutterPluginRegistrar));
+  UIImage *testImage = [self createOnePixelImage];
+  NSData *pngData = [self convertImageToPNGData:testImage];
+  XCTAssertNotNil(pngData);
+
+  FlutterStandardTypedData *typedData = [FlutterStandardTypedData typedDataWithBytes:pngData];
+
+  NSArray *iconData = @[ @"bytes", typedData, @"auto", @2, @[ @15.0, @15.0 ] ];
+
+  UIImage *resultImage = [instance extractIconFromData:iconData registrar:mockRegistrar];
+  XCTAssertNotNil(resultImage);
+}
+
+- (void)testExtractIconFromDataBytesNoScaling {
+  FLTGoogleMapMarkerController *instance = [[FLTGoogleMapMarkerController alloc] init];
+  NSObject<FlutterPluginRegistrar> *mockRegistrar =
+      OCMStrictProtocolMock(@protocol(FlutterPluginRegistrar));
+  UIImage *testImage = [self createOnePixelImage];
+  NSData *pngData = [self convertImageToPNGData:testImage];
+  XCTAssertNotNil(pngData);
+
+  FlutterStandardTypedData *typedData = [FlutterStandardTypedData typedDataWithBytes:pngData];
+
+  NSArray *iconData = @[ @"bytes", typedData, @"noScaling", @2 ];
+
+  UIImage *resultImage = [instance extractIconFromData:iconData registrar:mockRegistrar];
+  XCTAssertNotNil(resultImage);
+}
+
+- (void)testScaleSizeToInt {
```

Comment:
this seems to be a private function that shouldn't be exposed.

Reply:
These are exposed only for tests:

```
@interface FLTGoogleMapMarkerController (Test)
- (UIImage *)extractIconFromData:(NSArray *)iconData
                       registrar:(NSObject<FlutterPluginRegistrar> *)registrar;
- (CGSize)scaleSizeAndFloorToInt:(CGSize)size withFactor:(CGFloat)scale;
- (BOOL)isScalableWithScaleFactorFromSize:(CGSize)originalSize toSize:(CGSize)targetSize;
@end
```